wasm_src/ast_wrapper/ast_wrapper.cc

```diff
@@ -587,4 +587,158 @@ EMSCRIPTEN_KEEPALIVE float* fillTransformGrid(AstFrameSet* wcsInfo, double xMin,
 
     return out;
 }
+
+EMSCRIPTEN_KEEPALIVE AstFrameSet* makeSwappedFrameSet(AstFrameSet* originFrameSet, int dirAxis, int spectralAxis, int pixelZ, int nsample)
+{
+    astBegin;
+    int axisCount = 3;
+
+    if (astGetI(originFrameSet, "Nin") != axisCount || astGetI(originFrameSet, "Nout") != axisCount)
+    {
+        std::cerr << "Bad frame set!\n";
+        return nullptr;
+    }
+
+    if (dirAxis < 1 || dirAxis > axisCount || spectralAxis < 1 || spectralAxis > axisCount)
+    {
+        std::cerr << "Bad axis index!\n";
+        return nullptr;
+    }
+
+    AstMapping* originMap = static_cast<AstMapping*> astGetMapping(originFrameSet, AST__CURRENT, AST__BASE);
```

Comment:
Just curious, if we invert the order of AST_CURRENT and AST_BASE here, are we able to skip the astInvert() step at line 621?

Reply:
I think you are right. Thank you for the comment!